torch/_inductor/lowering.py

```diff
@@ -4384,6 +4383,110 @@ def adaptive_max_pool2d(x, output_size):
     return rv, ri
 
 
+fallback_fractional_max_pool2d = fallback_handler(
+    aten.fractional_max_pool2d.default, add_to_fallback_set=False
+)
+
+
+def _fractional_pooling_offsets(samples, in_sz, out_sz, kernel_sz, dim):
+    out_sz = out_sz[dim]
+    in_sz = in_sz[dim]
+    kernel_sz = kernel_sz[dim]
+    alpha = (in_sz - kernel_sz) / (out_sz - 1)
+    samples_loader = samples.make_loader()
+
+    def load(prefix, i):
+        sample = ops.indirect_indexing(
+            samples_loader([*prefix, dim]), size=3, check=False
+        )
+
+        seq_i = ops.index_expr(
+            sympy.floor((i + sample) * alpha) - sympy.floor(sample * alpha), torch.int64
+        )
+
+        mask = ops.lt(
+            ops.index_expr(i, torch.int64),
+            ops.sub(ops.index_expr(out_sz, torch.int64), ops.constant(1, torch.int64)),
+        )
+
+        return ops.where(mask, seq_i, in_sz - kernel_sz)
+
+    return load
+
+
+@register_lowering(aten.fractional_max_pool2d)
+def fractional_max_pool2d(x, kernel_size, output_size, random_samples):
+    x.realize_hint()
+    *batch, inp_h, inp_w = x.get_size()
+    inp_h = V.graph.sizevars.evaluate_static_shape(inp_h)
+    inp_w = V.graph.sizevars.evaluate_static_shape(inp_w)
```

Comment:
Do we need these?

Reply:
It looks like we do 1) The lowering for other pooling ops has this, and 2) testing it out it looks like that size can't be symbolic to use it as the `size` for an indirect indexing expression later.

Reply:
For 1), the other pooling ops may very well be improved as well lol.
2) What do you mean by "indrirect indexing cannot be symbolic"? See
https://github.com/pytorch/pytorch/blob/80900e2172f8b7b18ef44750a92bea27b51dd606/torch/_inductor/lowering.py#L2870